packages/blocks/src/store/selectors.js

```diff
@@ -89,9 +89,15 @@ export function getBlockVariations( state, blockName, scope ) {
 	if ( ! variations || ! scope ) {
 		return variations;
 	}
-	return variations.filter( ( variation ) => {
-		return ! variation.scope || variation.scope.includes( scope );
-	} );
+	// There are cases like `transform` where we want strict filtering of
+	// block variations and not return all the variations that do not have
+	// the `scope` set.
+	const variationFilterCb =
+		scope === 'transform'
```

Comment:
I don't like this implementation proposed because it depends on the value provided.

Based on the current API, it should be fine to assume that when the `scope` is set, and the variation has no `scope` set then the value used for operations should be `[ 'block', 'inserter' ]`.

```js
return variations.filter( ( variation ) => {
 	// For backward compatibility reasons, variation's scope defaults to `block` and `inserter` when not set.
 	return ( variation.scope || [ 'block', 'inserter' ] ).includes( scope );
 } );
```
It might be enough to code it like the above.